lms/djangoapps/completion_api/views.py

```diff
@@ -358,11 +363,45 @@ class CompletionDetailView(CompletionViewMixin, APIView):
     """
     # pylint: enable=line-too-long
 
-    def get(self, request, course_key):
+    def get(self, request, course_id):
         """
         Handler for GET requests.
         """
-        course_key = CourseKey.from_string(course_key)
-        progress = self.get_progress_queryset().get(course_id=course_key)
+        course_id = CourseKey.from_string(course_id)
+        progress = self.get_progress_queryset().get(course_id=course_id)
         completion = CourseCompletionFacade(progress)
         return Response(self.get_serializer()(completion, requested_fields=self.get_requested_fields()).data)
+
+
+class CompletionBlockUpdateView(APIView):
+    """
+    API view to mark blocks as completed.
+    This is a transitional implementation.
+    """
+
+    authentication_classes = (
+        authentication.OAuth2AuthenticationAllowInactiveUser,
+        authentication.SessionAuthenticationAllowInactiveUser
+    )
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, course_id, usage_id):
+        course_key = CourseKey.from_string(course_id)
+        try:
+            get_course(course_key)
+        except ValueError:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        content_key = UsageKey.from_string(usage_id)
+        try:
+            modulestore().get_item(content_key)
+        except ItemNotFoundError:
+            return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
We need two additional checks here: That the user is actually enrolled in the course, and that the `content_key` is from the same course as `course_key`. There should be examples of such checks in other APIs.